src/places.py

```diff
@@ -384,3 +401,41 @@ def _on_encrypted_eject_finished(self, encrypted, success):
             self._on_device_removed(None, encrypted)
         else:
             self.emit("failed", None)
+
+    def _find_place_by_path(self, listbox, path):
+        for place in listbox:
+            if place.path == path:
+                return place
+        return None
+
+    def _remove_bookmark(self, bookmark, path):
+        place = self._find_place_by_path(self._bookmarks_listbox, path)
+        if place is not None:
+            self._bookmarks_group.remove(place)
+            return True
+        return False
+
+    def _add_bookmark(self, bookmark, path):
+        place = self._find_place_by_path(self._bookmarks_listbox, path)
+        if place is None:
```

Comment:
you can simplify this whole thing like this:

```python
if not self._find_place_by_path(self._bookmarks_listbox, path):
    self._add_bookmark_place(path)
```

You don't need these returns.